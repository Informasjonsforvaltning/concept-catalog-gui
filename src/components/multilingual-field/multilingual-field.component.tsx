import React from 'react';
import { Field } from 'formik';

interface Props {
  name: string;
  label: string;
  component: Function;
  languages: any;
  showLabel?: boolean;
}

export const MultilingualField = ({ name, component, languages, label, showLabel = false }: Props): JSX.Element => (
  <>
    {showLabel && label && (
      <label className="fdk-form-label w-100 pl-2" htmlFor={name}>
        {label}
      </label>
    )}
    {languages.map(({ code, selected }) => {
      const isOnlyOneSelectedLanguage = languages.filter({ selected }).length === 1;

      return (
        selected && (
          <Field
            key={code}
            name={`${name}.${code}`}
            label={`${label}.${code}`}
            component={component}
            language={code}
            isOnlyOneSelectedLanguage={isOnlyOneSelectedLanguage}
          />
        )
      );
    })}
  </>
);
