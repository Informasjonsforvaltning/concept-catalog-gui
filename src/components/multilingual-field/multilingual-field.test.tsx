import React from 'react';
import { shallow } from 'enzyme';
import { MultilingualField } from './multilingual-field.component';

let defaultProps;
let wrapper;

beforeEach(() => {
  defaultProps = {
    name: null,
    label: null,
    component: null,
    languages: []
  };
  wrapper = shallow(<MultilingualField {...defaultProps} />);
});

test('should render MultilingualField correctly', () => {
  expect(wrapper).toMatchSnapshot();
});
