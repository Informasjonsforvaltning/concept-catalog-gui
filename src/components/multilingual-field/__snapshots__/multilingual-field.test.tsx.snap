// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render MultilingualField correctly 1`] = `<Fragment />`;
